pkg/shell/superuser.jsx

```diff
@@ -229,7 +229,7 @@ const UnlockDialog = ({ proxy, host }) => {
         <Modal isOpen
                position="top"
                variant="medium"
-               onClose={D.close}
+               onClose={methods ? (busy ? cancel : D.close) : cancel}
```

Comment:
I think this can just be `onClose={cancel || D.close}`.

The way `busy` and `cancel` are maintained and used is not very elegant in the existing code.  Maybe we can clean that up as well?  Hmm...

Reply:
From the quick read over the code I had, it seems that if `methods` is used `cancel` is a noop? but yeah we could likely simplify it. I have no idea what `methods` is, but I assume it's some custom way to gain privileges.

What are you ideas to clean up `busy` and `cancel`?

Reply:
> What are you ideas to clean up busy and cancel?

`cancel` should never be null and always be the correct function to call to close the dialog. (This is already the case, I believe, except for the initialization.)

`busy` is just used to disable the apply button and show the spinner.

`methods` should only be used in the big `if-else` train to set `title`, `title_icon`, `body`, and `footer`. Using it also in the construction of the `Modal` just increases complexity.

I'll push a FIXUP here, ok?

Reply:
LGTM!